src/boot/efi/boot.c

```diff
@@ -734,7 +743,9 @@ static BOOLEAN menu_run(
                         beep(idx_highlight + 1);
 
                 err = console_key_read(&key, timeout_remain > 0 ? 1000 * 1000 : UINT64_MAX);
-                if (err == EFI_TIMEOUT) {
+                if (err == EFI_NOT_READY)
+                        continue;
+                else if (err == EFI_TIMEOUT) {
                         timeout_remain--;
```

Comment:
maybe add an `asssert(timeout_remain > 0);` to clarify that we don't expect to see `EFI_TIMEOUT` if `timeout_remain == 0` holds